File: Backend/TopSys.TopConWeb.Domain/Entities/Visitas/Visita.cs
using System;
using System.Collections.Generic;
using TopSys.TopConWeb.Domain.Interfaces.Entities;

namespace TopSys.TopConWeb.Domain.Entities.Visitas
{
    public class Visita : IHasEndereco
    {
        public Usina Usina { get; set; }
        public int UsinaCodigo { get; set; }
        public int Numero { get; set; }
        public int Ano { get; set; }
        public VisitaTipo TipoVisita { get; set; }
        public int VisitaTipoCodigo { get; set; }
        public DateTime Data { get; set; }
        public TimeSpan HoraVisita { get; set; }
        public string Cliente { get; set; }
        public int DddTelefone { get; set; }
        public int Telefone { get; set; }
        public int DddCelular { get; set; }
        public int Celular { get; set; }
        public string Email { get; set; }
        public Vendedor Vendedor { get; set; }
        public int VendedorCodigo { get; set; }
        public string ObraNome { get; set; }
        public string EnderecoCep { get; set; }
        public string EnderecoLogradouro { get; set; }
        public int EnderecoNumero { get; set; }
        public string EnderecoComplemento { get; set; }
        public string EnderecoBairro { get; set; }
        public Municipio EnderecoMunicipio { get; set; }
        public int? EnderecoMunicipioCodigo { get; set; }
        public string ObservacaoInterna { get; set; }
        public VisitaContato ContatoPrincipal { get; set; }
        public VisitaContato ContatoSecundario { get; set; }
        public ICollection<VisitaLog> Logs { get; set; }

        public int LeadNumero { get; set; }
        public int LeadAno { get; set; }

    }
}

File: Frontend/src/app/classes/visita/visita.ts
import { Time } from "@angular/common";
import { Endereco, Municipio } from "../endereco/endereco";
import { Usina } from "../usina/usina";
import { Vendedor } from "../vendedor/vendedor";
import { VisitaContato } from "./visita-contato";
import { VisitaTipo } from "./visita-tipo";

export class Visita {

    usina: Usina;
    usinaCodigo: number;

    numero: number = 0;
    ano: number = 0;
    
    tipoVisita: VisitaTipo;
    visitaTipoCodigo: number = 0;
    
    data: Date = new Date();

    horaVisita: string = new Date().getHours().toString().padStart(2, '0') + ':' + new Date().getMinutes().toString().padStart(2, '0') + ':00';
    horaVisitaString: string = new Date().getHours().toString().padStart(2, '0') + new Date().getMinutes().toString().padStart(2, '0');

    cliente: string = '';
    dddTelefone: number = 0;
    telefone: number = 0;
    dddCelular: number = 0;
    celular: number = 0;
    email: string = '';

    vendedor: Vendedor;
    vendedorCodigo: number = 0;

    obraNome: string = ''
    observacaoInterna: string = '';
    
    endereco: Endereco = new Endereco();

    leadNumero: number = 0;
    leadAno: number = 0;

    contatoPrincipal: VisitaContato = new VisitaContato();
    contatoSecundario: VisitaContato = new VisitaContato();
}